server/plugin/api.go

```diff
@@ -984,10 +987,29 @@ func (p *Plugin) getYourAssignments(c *UserContext, w http.ResponseWriter, r *ht
 	result, _, err := githubClient.Search.Issues(c.Ctx, query, &github.SearchOptions{})
 	if err != nil {
 		c.Log.WithError(err).With(logger.LogContext{"query": query}).Warnf("Failed to search for assignments")
-		return
+		return nil
 	}
 
-	p.writeJSON(w, result.Issues)
+	return result.Issues
+}
+
+func (p *Plugin) getSidebarData(c *UserContext) *SidebarContent {
+	assignments := p.getYourAssignmentsData(c)
+	prs := p.getYourPrsData(c)
+	reviews := p.getReviewsData(c)
+	unreads := p.getUnreadsData(c)
+
+	return &SidebarContent{
+		Assignments: assignments,
+		PRS:         prs,
+		Reviews:     reviews,
+		Unreads:     unreads,
+	}
+}
+
+func (p *Plugin) getSidebarContent(c *UserContext, w http.ResponseWriter, r *http.Request) {
+	sidebarContent := p.getSidebarData(c)
```

Comment:
should not we need to check and handle if `sidebarContent` is `nil`

Reply:
I have not changed the already present logic for these APIs. I just grp those 3 APIs into 1. In case of errors, they are logging the error and the handling for nil is in the frontend code.

Reply:
It's a bad practice followed here that APIs and web app are tightly coupled, but let's skip this as it is beyond the scope of this ticket to fix all such error handling, you can log an issue though

Reply:
sure